usersimcrs/user_modeling/preference_model.py

```diff
@@ -75,171 +89,115 @@ class testable.
 
         # Store item and slot-value preferences separately.
         self._item_preferences = UserPreferences(self._user_id)
-        self._slotvalue_preferences = UserPreferences(self._user_id)
-
-        # Initialize item preferences by sampling from historical ratings.
-        self._initialize_item_preferences()
+        self._slot_value_preferences = UserPreferences(self._user_id)
 
-    def _initialize_item_preferences(self) -> None:
-        """Initializes the simulated user's preferences on items by sampling
-        from ratings of a historical user."""
-        # TODO Update, as currently all historical item preferences are copied;
-        # instead, there should only be a sample of them. Note that that'll make
-        # testing a bit tricky...
-        for item_id, rating in self._historical_ratings.get_user_ratings(
-            self._historical_user_id
-        ).items():
-            self._item_preferences.set_preference("ITEM_ID", item_id, rating)
+    def is_item_consumed(self, item_id: str) -> bool:
+        """Returns whether or not an item has been consumed by the user.
 
-    def _get_property_preferences(
-        self, property: str = "GENRE"
-    ) -> Dict[str, float]:
-        """Gets the property (e.g, genre) preferences based on rated items.
-
-        This is used to infer both item and slotvalue preference.
+        Args:
+            item_id: Item ID.
 
         Returns:
-            The averaged ratings based on rated items.
+            True if the item has been consumed (i.e., appears among the
+            historical ratings).
         """
-        property_ratings = defaultdict(list)
-        # Cache all the ratings for properties based on items, e.g.,
-        # {"action": [-1, -1, -1]}.
-        for (
-            hist_item_id,
-            rating,
-        ) in self._item_preferences.get_preferences("ITEM_ID").items():
-            # TODO: handling all properties generally by creating a new issue
-            for genre in self._item_collection.get_item(
-                hist_item_id
-            ).get_property(property):
-                property_ratings[genre].append(rating)
-        property_preferences = dict()
-        # Calculate the averaged rating for each genre.
-        # TODO: use property instead of genre and make property a parameter of
-        # property_ratings
-        for genre, rating_list in property_ratings.items():
-            property_preferences[genre] = round(
-                sum(rating_list) / len(rating_list), ndigits=1
-            )
-        return property_preferences
+        return item_id in self._historical_ratings
```

Comment:
`self._historical_ratings` is an instance of `Ratings`, hence, I have a dount that using `in` will behave as expected.

Reply:
Indeed, thanks for catching this.